frontend/src/host_orchestrator/orchestrator/instancemanager.go

```diff
@@ -455,24 +477,49 @@ func (e *cvdCommandExecErr) Error() string {
 
 func (e *cvdCommandExecErr) Unwrap() error { return e.err }
 
+type cvdCommandTimeoutErr struct {
+	args []string
+}
+
+func (e *cvdCommandTimeoutErr) Error() string {
+	return fmt.Sprintf("cvd execution with args %q timed out", strings.Join(e.args, " "))
+}
+
 func (c *cvdCommand) Run() error {
 	// Makes sure cvd server daemon is running before executing the cvd command.
 	if err := c.startCVDServer(); err != nil {
 		return err
 	}
+	done := make(chan error)
+	var b bytes.Buffer
 	cmd := buildCvdCommand(c.execContext, c.androidHostOut, c.home, c.cvdBin, c.args...)
-	stdoutStderr, err := cmd.CombinedOutput()
+	cmd.Stdout = &b
+	cmd.Stderr = &b
+	err := cmd.Start()
 	if err != nil {
-		msg := "`cvd` execution failed with combined stdout and stderr:\n" +
-			"############################################\n" +
-			"## BEGIN \n" +
-			"############################################\n" +
-			"\n%s\n\n" +
-			"############################################\n" +
-			"## END \n" +
-			"############################################\n"
-		log.Printf(msg, string(stdoutStderr))
-		return &cvdCommandExecErr{c.args, string(stdoutStderr), err}
+		return err
+	}
+	go func() { done <- cmd.Wait() }()
+	var timeoutCh <-chan time.Time
+	if c.timeout != 0 {
+		timeoutCh = time.After(c.timeout)
+	}
+	select {
+	case err := <-done:
+		if err != nil {
+			msg := "`cvd` execution failed with combined stdout and stderr:\n" +
+				"############################################\n" +
+				"## BEGIN \n" +
+				"############################################\n" +
+				"\n%s\n\n" +
+				"############################################\n" +
+				"## END \n" +
+				"############################################\n"
+			log.Printf(msg, b.String())
+			return &cvdCommandExecErr{c.args, b.String(), err}
+		}
+	case <-timeoutCh:
+		return &cvdCommandTimeoutErr{c.args}
```

Comment:
IIUC this marks the Run as failed but doesn't cancel execution of the command. Meaning a command can actually succeed after the timeout is reached and reported?

Reply:
yes, I planned to cancel the whole process in a different PR as is not as straightforward as killing the process represented by `cmd`, I also need to kill all the process spawned by cvd server for this action.